core/src/main/java/com/inrupt/client/core/DefaultClient.java

```diff
@@ -126,6 +128,25 @@ Request upgradeRequest(final Request request, final Credential token) {
         return builder.build();
     }
 
+    @SuppressWarnings("unchecked")
+    static <T, R extends Throwable> Response<T> recover(final Response<T> res, final Throwable err) throws R {
+        if (err != null) {
+            if (err instanceof ClientHttpException) {
+                final ClientHttpException ex = (ClientHttpException) err;
+                if (ex.getStatusCode() == UNAUTHORIZED) {
+                    return new NoBodyResponse<T>(ex.getUri(), ex.getStatusCode(), ex.getHeaders());
+                }
+            } else if (err.getCause() instanceof ClientHttpException) {
```

Comment:
The error and its cause are inpected, but the hierarchy is not traversed further.
I'm sure this is accurate vis a vis our architecture, but it feels off.
I think an explanation with inline comments would be good.